OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -189,21 +186,21 @@ ElementPropertiesDialog::ElementPropertiesDialog(Element *pComponent, QWidget *p
           QDoubleValidator *pDoubleValidator = new QDoubleValidator(this);
           pParameterLineEdit->setValidator(pDoubleValidator);
           double value;
-          if (modelInstantiated && (status = OMSProxy::instance()->getReal(nameStructure, &value))) {
```

Comment:
Get rid of these extra parenthesis as well.

Reply:
Qt is giving warning if i remove that extra paranthesis, it suggests to keep the paranthesis

Reply:
The parentheses are needed since it's an assignment, many compilers will complain otherwise since it's assumed that the user accidentally used `=` instead of `==`.

Reply:
Aha since its an assignment at the same time. Keep it then.

Reply:
Just personal preferences but I think this should be either

```c
status = OMSProxy::instance()->getReal(nameStructure, &value);
if (status) {
  pParameterLineEdit->setText(QString::number(value));
}
```

OR

```c
if(OMSProxy::instance()->getReal(nameStructure, &value)) {
  pParameterLineEdit->setText(QString::number(value));
  textSet= true;  // instead of status
}
..

if (!textSet) {
  pParameterLineEdit->setPlaceholderText("unknown");
}
```

Reply:
Yes, don't put assignments in the if condition. Unnecessarily complicated to read and easy to misread.